src/node_nanomsg.cc

```diff
@@ -199,21 +199,24 @@ NAN_METHOD(Shutdown) {
 NAN_METHOD(Send) {
   NanScope();
 
-  int s = args[0]->Uint32Value();
+  int flags = args[2].integer;
+  std::string *input;
 
-  if (!node::Buffer::HasInstance(args[1])) {
-    return NanThrowError("second Argument must be a Buffer.");
-  }
+  if (node::Buffer::HasInstance(args[1]->ToObject())) {
 
-  Local<Object> obj = args[1]->ToObject();
-  char *odata = node::Buffer::Data(obj);
-  size_t odata_len = node::Buffer::Length(obj);
-  int flags = args[2]->Uint32Value();
+    v8::Handle<v8::Object> object = args[1]->ToObject();
 
-  // Invoke nanomsg function.
-  int ret = nn_send(s, odata, odata_len, flags);
+    const char *data = node::Buffer::Data(object);
+    input = new std::string(data, node::Buffer::Length(object));
```

Comment:
If a node::buffer::Data object can tell you its length, then you can explicitly set the last char (len - 1) to `'\0'`.  No need to construct a std::string just to zero out the last byte.  Instead, what you should do is create a `char []`, whose length is node::Buffer::Data's length + 1, memcpy the data, then explicitly set the last byte to `'\0'`.  This will overflow the stack for messages larger than 5 MB on most OS', so maybe a length check and a warning or heap allocation.